gateway/mw_oas_validate_request.go

```diff
@@ -52,7 +52,13 @@ func (k *ValidateRequest) ProcessRequest(w http.ResponseWriter, r *http.Request,
 		return nil, http.StatusOK
 	}
 
-	router, err := gorillamux.NewRouter(&k.Spec.OAS.T)
+	// replacing servers object to just have listen path so that router.FindRoute(r) will not fail with strict hostname check
+	oasSpec := k.Spec.OAS.T
+	oasSpec.Servers = openapi3.Servers{
+		{URL: k.Spec.Proxy.ListenPath},
+	}
+
+	router, err := gorillamux.NewRouter(&oasSpec)
```

Comment:
Can we create an oas.NewRouter(spec) func. Since we will fork apidef, having the openapi dependency imported there may avoid some problems :)

Reply:
oas.Newrouter to just wrap this call?

Reply:
Just to invoid the import of openapi3 into gateway. We will need to match/update the dependency with apidef, just having the openapi3 import under apidef oas.

An option would just be to add an oas.PrepareHost(*OAS, string) *OAS, so we don't have a mux dependency in apidef;

This avoids syncing the openapi3 dependency versions from apidef into go.mod for consumers (gw, dash...)

Reply:
Figured out we have a few imports already, so we are going to address this separately. :)